app/src/view/DatePicker.js

```diff
@@ -0,0 +1,161 @@
+    datepicker: {
+      backgroundColor: theme.palette.background.paper,
+      borderColor: borderColor,
+      borderTopColor: 'transparent',
+      border: 'solid 1px',
+      borderRadius: theme.shape.borderRadius,
+      padding: '18.5px 14px',
+      color: theme.palette.text.primary,
+      fontSize: theme.typography.fontSize,
+      lineHeight: '18px',
+      width: '100%',
+      '&:hover': {
+        borderColor: theme.palette.text.primary,
+        borderTopColor: 'transparent',
+      },
+    },
+    span: {
+      position: 'absolute',
+      top: '0',
+      left: '0',
+      display: 'flex',
+      width: '100%',
+      fontSize: '10px',
+      lineHeight: '15px',
+      '&:before,&:after': {
+        content: '""',
+        display: 'block',
+        boxSizing: 'border-box',
+        marginTop: '6px',
+        borderTop: 'solid 1px',
+        borderColor: borderColor,
+        minWidth: '10px',
+        height: '8px',
+      },
+      '&:hover::before,&:hover::after': {
+        borderColor: theme.palette.text.primary,
+      },
+      '&:before': {
+        marginRight: '4px',
+        borderLeft: 'solid 1px transparent',
+        borderRadius: '4px 0',
+      },
+      '&:after': {
+        flexGrow: '1',
+        marginLeft: '4px',
+        borderRight: 'solid 1px transparent',
+        borderRadius: '0 4px 4px 0',
+        height: '57px',
+      },
+    },
+    '@global': {
+      '#datepickers .react-datepicker__header, #datepickers .react-datepicker__current-month, #datepickers .react-datepicker__day-name, #datepickers .react-datepicker-time__header': {
+        backgroundColor: theme.palette.background.default,
+        color: theme.palette.text.primary,
+        borderColor: borderColor,
+      },
+      '#datepickers .react-datepicker, #datepickers .react-datepicker__day, #datepickers .react-datepicker__time, #datepickers .react-datepicker__time-container': {
+        backgroundColor: theme.palette.background.paper,
+        color: theme.palette.text.primary,
+        borderColor: borderColor,
+      },
+      '#datepickers .react-datepicker__time-list-item': {
+        height: 'auto',
+        padding: '7px 10px',
+      },
+      '#datepickers .react-datepicker__day--selected, #datepickers .react-datepicker__time-list-item--selected': {
+        backgroundColor: theme.palette.primary.main,
+        color: theme.palette.primary.contrastText,
+      },
+      '#datepickers .react-datepicker__day:hover, #datepickers .react-datepicker__time-list-item:hover': {
+        backgroundColor: theme.palette.background.default,
+      },
+    },
+  };
+});
+
+const Portal = ({children}) => {
+  const portalRoot = document.getElementById("portal");
+  const el = document.createElement("div");
+  el.id = "datepickers";
+
+  if (!portalRoot) {
+    console.error("No portal root found");
+    return;
+  }
+
+  React.useEffect(() => {
+    portalRoot.appendChild(el);
+    return () => {
+      portalRoot.removeChild(el);
+      return;
```

Comment:
The `return` is for another flow error. It seems really awkward to me to have to explicitly return nothing rather than returning a `HTMLDivElement` that would be ignored.

```
     Error ------------------------------------------------------------------------------------- src/view/DatePicker.js:93:18

Cannot call `useEffect` with function bound to `create` because `HTMLDivElement` [1] is incompatible with undefined [2]
in the return value of the return value.

   src/view/DatePicker.js:93:18
    93|     return () => portalRoot.removeChild(el);
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^

References:
   /tmp/flow/flowlib_e4a0f5a/dom.js:764:68
   764|   createElement(tagName: 'div', options?: ElementCreationOptions): HTMLDivElement;
                                                                           ^^^^^^^^^^^^^^ [1]
   /tmp/flow/flowlib_e4a0f5a/react.js:316:47
   316|   declare type MaybeCleanUpFn = void | (() => void);
                                                      ^^^^ [2]
```

Reply:
So this is because single-line blockless arrow functions implicitly return a value.
`() => portalRoot.removeChild(el)`  returns a value, but `() => { portalRoot.removeChild(el); }` returns undefined.

Reply:
The braces are all you need to prevent a return value: `return () => { portalRoot.removeChild(el); };`

Reply:
Of course, makes sense, thanks!